extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -215,34 +239,32 @@ export class AmpAutocomplete extends AMP.BaseElement {
       );
     }
 
-    if (
-      this.templates_.hasTemplate(this.element, 'template, script[template]')
-    ) {
-      this.templateElement_ = this.templates_.findTemplate(
-        this.element,
-        'template, script[template]'
+    this.isSsr_ = this.ssrTemplateHelper_.isSupported();
+    this.hasTemplate_ = this.templates_.hasTemplate(
+      this.element,
+      'template, script[template]'
+    );
+    if (this.isSsr_) {
+      userAssert(
```

Comment:
Not sure I follow here, why does SSR being supported by the viewer imply that the element must have a template? Does SSR supported by the viewer imply that it must be used? I think that should be made more clear. The variable here, `isSsr_`, seems to conflict with the method it comes from `isSupported()`, which is implying optional, but not required.

Can you think of a case where SSR is supported, but not required?

Reply:
I agree the method is confusingly named. I clarified the case you mention with @choumx and IIUC the intention is that whenever `isSupported()` is true, server side rendering is expected and required. For now I can add a comment to that effect here, but perhaps it would be valuable to rename that method later on?

Reply:
I agree. After this lands I'll rename it to something more like `isSsrEnabled`